src/main/java/com/exactpro/th2/fix/client/fixBean/FixBean.java

```diff
@@ -42,40 +45,80 @@ public StringBuilder toConfig(String sectionName) { //todo mb getting fixBean as
+            sb.append("SocketConnectPort=").append(socketConnectPort).append(System.lineSeparator());
+        if (senderCompID != null) sb.append("SenderCompID=").append(senderCompID).append(System.lineSeparator());
+        if (senderSubID != null) sb.append("SenderSubID=").append(senderSubID).append(System.lineSeparator());
+        if (senderLocationID != null)
+            sb.append("SenderLocationID=").append(senderLocationID).append(System.lineSeparator());
+        if (targetCompID != null) sb.append("TargetCompID=").append(targetCompID).append(System.lineSeparator());
+        if (targetSubID != null) sb.append("TargetSubID=").append(targetSubID).append(System.lineSeparator());
+        if (targetLocationID != null)
+            sb.append("TargetLocationID=").append(targetLocationID).append(System.lineSeparator());
+        if (dataDictionary != null)
+            sb.append("DataDictionary=").append(dataDictionary).append(System.lineSeparator());
+
 
         return sb;
     }
 
+    @Override
+    public String toString() {
+        return new ToStringBuilder(this, ToStringStyle.SHORT_PREFIX_STYLE)
+                .append("ApplicationId", applicationID)
+                .append("FileStorePath", fileStorePath)
+                .append("FileLogPath", fileLogPath)
+                .append("ConnectionType", connectionType)
+                .append("ReconnectInterval", reconnectInterval)
+                .append("HeartBtInt", heartBtInt)
+                .append("UseDataDictionary", useDataDictionary)
+                .append("ValidateUserDefinedFields", validateUserDefinedFields)
+                .append("ValidateIncomingMessage", validateIncomingMessage)
+                .append("RefreshOnLogon", refreshOnLogon)
+                .append("NonStopSession", nonStopSession)
+                .append("BeginString", beginString)
+                .append("SocketConnectHost", socketConnectHost)
+                .append("SocketConnectPort", socketConnectPort)
+                .append("SenderCompID", senderCompID)
+                .append("SenderSubID", senderSubID)
+                .append("SenderLocationID", senderLocationID)
+                .append("TargetCompID", targetCompID)
+                .append("TargetSubID", targetSubID)
+                .append("TargetLocationID", targetLocationID)
+                .append("DataDictionary", dataDictionary)
+                .append("SessionQualifier", sessionQualifier)
+                .append("SessionAlias", sessionAlias)
+                .toString();
+    }
+
+    public void setTargetCompID(String targetCompID) {
+        if (targetCompID == null || targetCompID.equals(""))
+            throw new IllegalArgumentException("TargetCompID must not be null or blank.");
+        this.targetCompID = targetCompID;
+    }
+
     public void setSocketConnectPort(long socketConnectPort) {
+        if (socketConnectPort == 0) throw new IllegalArgumentException("SocketConnectPort must not be null or blank.");
```

Comment:
We should check that it's in acceptable range - 0..65535, but the lower bound is usually raised to 1024 since ports below it are usually used by system services

Reply:
in this case, is it a better solution to make it a string?

Reply:
> in this case, is it a better solution to make it a string?

Why?